unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3475 @@
+
+    ASSERT_EQ(get_id_of_layer_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_get_user_data_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 0);
+}
+
+/** ================================================================================================
+ * @test_id             send_layer_prop_hasEvents
+ * @brief               Test case of send_layer_prop() where event mask of ivi layer is the change of all properties
+ * @test_procedure Steps:
+ *                      -# Set event mask of ivi layer is the change of all properties
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Calling the send_layer_prop()
+ *                      -# Verification point:
+ *                         +# get_id_of_layer() must be called once time
+ *                         +# wl_resource_get_user_data() must be called once time
+ *                         +# wl_resource_post_event() must be called 4 times
+ */
+TEST_F(ControllerTests, send_layer_prop_hasEvents)
+{
+    m_layoutLayerProperties[0].event_mask = IVI_NOTIFICATION_OPACITY | IVI_NOTIFICATION_SOURCE_RECT |
+                                            IVI_NOTIFICATION_DEST_RECT | IVI_NOTIFICATION_VISIBILITY;
+
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)mp_iviController, 1);
+
+    send_layer_prop(&mp_iviLayer[0]->property_changed, nullptr);
+
+    ASSERT_EQ(get_id_of_layer_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_get_user_data_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 4);
+}
+
+/** ================================================================================================
+ * @test_id             surface_event_create_idDifferentBkgnSurfaceId
+ * @brief               Test case of surface_event_create() where id_surface different bkgnd_surface_id
+ * @test_procedure Steps:
+ *                      -# Set id_surface and bkgnd_surface_id
+ *                      -# Mocking the surface_get_weston_surface() does return an object
+ *                      -# Calling the surface_event_create()
+ *                      -# Verification point:
+ *                         +# get_id_of_surface() must be called once time
+ *                         +# get_properties_of_surface() must be called once time
+ *                         +# wl_list_init() must be called once time
+ *                         +# surface_get_weston_surface() must be called once time
+ *                         +# wl_list_insert() must be called 2 times
+ *                         +# surface_add_listener() must be called once time
+ *                         +# wl_resource_post_event() must be called {MAX_NUMBER} times
+ *                         +# The result output should same with prepare data
+ *                         +# Free resources are allocated when running the test
+ */
+TEST_F(ControllerTests, surface_event_create_idDifferentBkgnSurfaceId)
```

Comment:
rename to surface_event_create_idDifferentBkgnSurfaceId -> surface_event_create_normalId